### PR TITLE
Value management in programs

### LINKED ISSUE #644: Account value balance in contracts
### Problem to Solve

At the moment, programs are not able to check their total value balance.

### Possible Solution

Pass this info into `core-processor` with `Dispatch` or `ExecutionSettings`. For the node side include real balances, for in-memory testing - emulate them.

Provide `gstd` and `gcore` api for it (e.g. `exec::value_available()`).

### PR DESCRIPTION
**Pull request by breathx**
Resolves #644 and partially #649.

Left todo-s in code, related to 644 issue now will be covered in #695 to prevent making this pr so huge.

**Release notes**: _to be written..._

@gear-tech/dev

### CODE REVIEW

File: gtest/src/manager.rs [hunk 1]
```diff
@@ -348,7 +351,7 @@ impl JournalHandler for ExtManager {
         }
     }
     fn update_nonce(&mut self, program_id: ProgramId, nonce: u64) {
-        if let Some((Program::Core(prog), _)) = self.programs.get_mut(&program_id) {
+        if let Some((Program::Core(prog), _, _)) = self.actors.get_mut(&program_id) {
```
Comment (shamilsan) [inline gtest/src/manager.rs:354]: Two wildcards may be condensed if we don't care about their count.

```suggestion
        if let Some((Program::Core(prog), ..)) = self.actors.get_mut(&program_id) {
```

https://rust-lang.github.io/rustfmt/?version=v2.0.0-rc.2&search=#condense_wildcard_suffixes

File: core-processor/src/ext.rs [hunk 2]
```diff
@@ -44,7 +46,9 @@ pub struct Ext {
     pub block_info: BlockInfo,
     /// Allocations config.
     pub config: AllocationsConfig,
-    /// Is lazy-pages mode enabled ?
+    /// Account existence deposit
+    pub existence_deposit: u128,
```
Comment (NikVolf) [inline core-processor/src/ext.rs:50]: existential_deposit

File: gear-test/src/proc.rs [hunk 3]
```diff
@@ -17,6 +17,8 @@ use std::{
     time::{SystemTime, UNIX_EPOCH},
 };
 
+pub const EXISTENCE_DEPOSIT: u128 = 500;
```
Comment (NikVolf) [inline gear-test/src/proc.rs:20]: EXISTENTIAL_DEPOSIT

**Review by NikVolf — APPROVED** [review]
other than naming issue

**Pull request merged**